From e1aa8bd2d632f19a4402abeb1cde6a444a4a2ab5 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@freescale.com>
Date: Tue, 10 Nov 2015 19:05:02 +0800
Subject: [PATCH 0667/5242] MLK-11844-1 ARM: dts: imx6sx: reserve DDR space
 for M4 use case

commit  c08c559db771915f67e6323ed239c5dcddea6bde from
https://source.codeaurora.org/external/imx/linux-imx.git

FreeRTOS/M4 has DDR use case, M4 can only execute on DDR
space lower than 0xa0000000, Linux kernel will reserve 1M
(0x9ff00000 - 0x9fffffff) for it, below are the DDR usage:

0x80000000 - 0x9fefffff: Linux kernel;
0x9ff00000 - 0x9fffffff: FreeRTOS;
0xa0000000 - 0xbfefffff: Linux kernel;
0xbff00000 - 0xbfffffff: RPMSG shared memory;
0xc0000000 - 0xffffffff: Linux kernel if boards has 2GB DDR.

Also, remove 'reg' property which is unused.

Signed-off-by: Anson Huang <Anson.Huang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto-m4.dts |    8 ++++++++
 arch/arm/boot/dts/imx6sx-sdb-m4.dts       |    4 ++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
index 1273a28..159e907 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto-m4.dts
@@ -8,6 +8,14 @@
 
 #include "imx6sx-sabreauto.dts"
 
+/{
+	memory {
+		linux,usable-memory = <0x80000000 0x1ff00000>,
+				<0xa0000000 0x1ff00000>,
+				<0xc0000000 0x40000000>;
+	};
+};
+
 /*
  * The flollowing modules are conflicting with M4, disable them when m4
  * is running.
diff --git a/arch/arm/boot/dts/imx6sx-sdb-m4.dts b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
index 54a2ab4..d96544d 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
@@ -10,8 +10,8 @@
 
 /{
 	memory {
-		linux,usable-memory = <0x80000000 0x3ff00000>;
-		reg = <0x80000000 0x40000000>;
+		linux,usable-memory = <0x80000000 0x1ff00000>,
+				<0xa0000000 0x1ff00000>;
 	};
 };
 
-- 
1.7.9.5

